app/components/taxonomy-top-list.js

```diff
@@ -18,18 +19,35 @@ import Analytics from '../mixins/analytics';
  */
 export default Ember.Component.extend(Analytics, {
     theme: Ember.inject.service(),
-    // jscs:disable disallowQuotedKeysInObjects
-    subjectTooltips: {
-        'Architecture': 'Architectural Engineering, Construction Engineering, Environmental Design, Interior Architecture, Landscape Architecture',
-        'Arts and Humanities': 'Fine Arts, History, Music, Philosophy, Religion',
-        'Business': 'Accounting, Finance and Financial Management, Human Resource Management, Marketing, Taxation',
-        'Education': 'Curriculum and Instruction, Educational Administration and Supervision, Educational Leadership, Higher Education, Liberal Studies',
-        'Engineering': 'Biomedical Engineering and Bioengineering, Chemical Engineering, Civil and Environmental Engineering, Electrical and Computer Engineering, Mechanical Engineering',
-        'Law': 'Civil Law, Criminal Law, Legislation, State and Local Government Law, Supreme Court of the United States',
-        'Life Sciences': 'Agriculture, Genetics and Genomics, Microbiology, Physiology, Zoology',
-        'Medicine and Health Sciences': 'Anatomy, Diseases, Medical Sciences, Public Health, Veterinary Medicine',
-        'Physical Sciences and Mathematics': 'Chemistry, Computer Sciences, Earth Sciences, Physics, Statistics and Probability',
-        'Social and Behavioral Sciences': 'Anthropology, Economics, Political Science, Psychology, Sociology',
+    getSubjectChildren(parents, provider) {
+        return provider.query('taxonomies', {
+                    filter: {
+                        parents
+                    },
+                    page: {
+                        size: 4  //number of subjects to show-up in each tooltip.
+                    }
+                })
+            .then(results =>
+                results.map(result => result.get('text'))
+            )
+            .then(results =>
```

Comment:
No need to put another `.then` here. You can chain the result from `results.map`. A `.then` is only necessary if the previous statement returns a promise.